presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayEqualOperator.java

```diff
@@ -61,4 +62,64 @@ public static boolean equals(
         }
         return true;
     }
+
+    @TypeParameter("E")
```

Comment:
I think for type with constraints, we use `T` instead of `E`. Is this an convention? @haozhun

Reply:
I used `E` to be consistent with the existing `equals` function in this class.

Reply:
No, the name is irrelevant. There’s no explicit convention for it.